packages/nextjs/pages/game/[id].tsx

```diff
@@ -127,8 +128,10 @@ function GamePage() {
   useEffect(() => {
     let isHiiddenChars;
 
-    if (rolled && rolledResult.length > 0 && game?.hiddenChars) {
-      isHiiddenChars = compareResult(rolledResult, game?.hiddenChars);
+    if (rolled && rolledResult.length > 0 && game?.hiddenPrivateKey) {
+      const pk: `0x{string}` = `0x${rolledResult.join("")}${game?.hiddenPrivateKey.replaceAll("*", "")}` as `0x{string}`;
+      const account = privateKeyToAccount(pk);
```

Comment:
This is the key. Check if the private key is found, create the account, and check the address.